test/libyul/yulOptimizerTests/fullSuite/abi2.yul

```diff
@@ -1059,53 +1072,89 @@
+//         {
+//             revert(value4, value4)
+//         }
+//         if gt(add(add(_1092, abi_decode_length_15_689), 0x20), dataEnd_56)
+//         {
+//             revert(value4, value4)
+//         }
+//         value2_59 := add(_1092, 0x20)
+//         value3 := abi_decode_length_15_689
+//         value4 := cleanup_revert_t_enum$_Operation_$1949(calldataload(add(headStart_55, 96)))
 //     }
-//     let _1145 := 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
-//     sstore(_3, and(calldataload(_1017), _1145))
-//     let _1019 := mload(8)
-//     let _1021 := mload(7)
-//     let abi_decode_value2_59 := _3
-//     let abi_decode_value3 := _3
-//     if slt(sub(_1019, _1021), 128)
+//     function abi_decode_tuple_t_contract$_Module_$1038t_contract$_Module_$1038(headStart_154, dataEnd_155) -> value0_156, value1_157
 //     {
-//         revert(_3, _3)
+//         if slt(sub(dataEnd_155, headStart_154), 64)
+//         {
+//             revert(0, 0)
+//         }
+//         let cleanup_as__1127 := 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
+//         value0_156 := and(calldataload(add(headStart_154, 0)), cleanup_as__1127)
+//         value1_157 := and(calldataload(add(headStart_154, 32)), cleanup_as__1127)
 //     }
-//     let abi_decode_offset_62 := calldataload(add(_1021, 64))
-//     if gt(abi_decode_offset_62, 0xffffffffffffffff)
+//     function abi_encode_t_address_to_t_address(value_164, pos)
 //     {
-//         revert(_3, _3)
+//         mstore(pos, and(value_164, 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF))
 //     }
-//     abi_decode_value2_59, abi_decode_value3 := abi_decode_t_bytes_calldata_ptr(add(_1021, abi_decode_offset_62), _1019)
-//     let abi_decode_value4_1063 := cleanup_revert_t_enum$_Operation_$1949(calldataload(add(_1021, 96)))
-//     sstore(_1, and(calldataload(_1021), _1145))
-//     sstore(_1, calldataload(add(_1021, 32)))
-//     sstore(_1, _3)
-//     sstore(_1, _3)
-//     sstore(_1, abi_decode_value4_1063)
-//     function abi_decode_t_bytes_calldata_ptr(offset_12, end_13) -> arrayPos_14, length_15
+//     function abi_encode_t_bytes32_to_t_bytes32(value_171, pos_172)
 //     {
-//         if iszero(slt(add(offset_12, 0x1f), end_13))
-//         {
-//             revert(0, 0)
-//         }
-//         let length_15_669 := calldataload(offset_12)
-//         length_15 := length_15_669
-//         if gt(length_15_669, 0xffffffffffffffff)
-//         {
-//             revert(0, 0)
-//         }
-//         arrayPos_14 := add(offset_12, 0x20)
-//         if gt(add(add(offset_12, length_15_669), 0x20), end_13)
+//         mstore(pos_172, value_171)
+//     }
+//     function abi_encode_t_enum$_Operation_$1949_to_t_uint8(value_181, pos_182)
+//     {
+//         if iszero(lt(value_181, 3))
 //         {
-//             revert(0, 0)
+//             invalid()
 //         }
+//         mstore(pos_182, value_181)
+//     }
+//     function abi_encode_t_uint256_to_t_uint256(value_235, pos_236)
```

Comment:
This function should be inlined.

Reply:
Ah, I think it is not inlined because of the "we do not inline into larger functions" rule. This should be relaxed in the case where the function's body is a single function call to another function.